infra/testing/sw-env.js

```diff
@@ -11,21 +11,158 @@
  limitations under the License.
  */
 
-const mockFetch = require('./mock-fetch');
 const {IDBFactory, IDBKeyRange} = require('shelving-mock-indexeddb');
 const makeServiceWorkerEnv = require('service-worker-mock');
+const mockFetch = require('./mock-fetch');
+
+Object.assign(global, makeServiceWorkerEnv());
+global.self = global;
```

Comment:
Why are you assigning global to `global.self`? makeServiceWorkerEnv has a `self` property which you'll now we overriding with everything node has.

Reply:
This was necessary to handle the mocks we're adding later. I could try doing the global assignment after adding the mocks to see if that works. I remember there being an issue that this solve, but I can't remember all the various things I tried.